homeassistant/components/climate/evohome.py

```diff
@@ -104,6 +104,25 @@
     """Create the evohome Controller, and its Zones, if any."""
     evo_data = hass.data[DATA_EVOHOME]
 
+    client = evo_data['client']
+    loc_idx = evo_data['params'][CONF_LOCATION_IDX]
+
+    # evohomeclient has exposed no means of accessing non-default location
+    # (i.e. loc_idx > 0) other than using a protected member, such as below
+    tcs_obj_ref = client.locations[loc_idx]._gateways[0]._control_systems[0]    # noqa E501; pylint: disable=protected-access
+
+    evo_data['parent'] = EvoController(evo_data, client, tcs_obj_ref)
```

Comment:
Since `evo_data` is a shared dict we are not allowed to store entities in it. Why do we need to store the entities in the first place?

Reply:
Yeah, it was a bad idea.  It was to solve a problem I don't have yet!

Reply:
I have removed it - the issue will be when I add a water_heater device to the component (in a subsequent PR).